flash/data/process.py

```diff
@@ -217,30 +223,16 @@ def __init__(
         self._callbacks: List[FlashCallback] = []
         self._default_collate: Callable = default_collate
 
-    @property
-    def default_train_transforms(self) -> Optional[Dict[str, Callable]]:
-        return None
-
-    @property
-    def default_val_transforms(self) -> Optional[Dict[str, Callable]]:
-        return None
+    def _resolve_transforms(self, running_stage: RunningStage) -> Optional[Dict[str, Callable]]:
+        from flash.data.data_pipeline import DataPipeline
 
-    @property
-    def default_test_transforms(self) -> Optional[Dict[str, Callable]]:
-        return None
+        resolved_function = getattr(
+            self, DataPipeline._resolve_function_hierarchy("default_transforms", self, running_stage, Preprocess)
+        )
 
-    @property
-    def default_predict_transforms(self) -> Optional[Dict[str, Callable]]:
-        return None
-
-    @property
-    def transforms(self) -> Dict[str, Optional[Dict[str, Callable]]]:
-        return {
-            "train_transform": self.train_transform,
-            "val_transform": self.val_transform,
-            "test_transform": self.test_transform,
-            "predict_transform": self.predict_transform,
-        }
+        with CurrentRunningStageFuncContext(running_stage, "default_transforms", self):
```

Comment:
I think only CurrentRunningStageContext should be fine. Do we need current_fn = `default_transforms` ?